sympy/matrices/expressions/matpow.py

```diff
@@ -4,18 +4,19 @@
 from sympy.core.sympify import _sympify
 from sympy.core.compatibility import range
 from sympy.matrices import MatrixBase
-from sympy.core import S
+from sympy.core import S, Symbol
 
 
 class MatPow(MatrixExpr):
 
-    def __new__(cls, base, exp):
+    def __new__(cls, base, exp, evaluate=False):
```

Comment:
> Maybe it should remain unevaluated if the matrix power algorithm fails. I don't see the problem in evaluating it.

But this would change the existing system to a great extent. Many Sympy users expect MatPow to return an unevaluated object, only to be evaluated when `doit()` is used.

Example : 
```
>>A = Matrix([[1, 2],[3, 4]])
>>MatPow(A, 2)
```
Expected Behaviour : 
```
Matrix([
[1, 2],
[3, 4]])**2
```
New Behaviour : 
```
Matrix([
[ 7, 10],
[15, 22]])
```
 I did this change and a lot of existing unit tests failed.